machina/apps/forum_conversation/views.py

```diff
@@ -498,7 +498,7 @@ def form_valid(self, post_form, attachment_formset, poll_option_formset, **kwarg
         valid = super().form_valid(
             post_form, attachment_formset, poll_option_formset=poll_option_formset, **kwargs)
 
-        if save_poll_option_formset:
+        if save_poll_option_formset and poll_option_formset.is_valid():
```

Comment:
I don't think we should call `is_valid()` from within the `form_valid` method: at this point the validations has already been performed. Instead I think you could probably make use of the `poll_options_validated` option that is passed to this method in order to fix this issue.

Probably something along those lines:

```diff
diff --git a/machina/apps/forum_conversation/views.py b/machina/apps/forum_conversation/views.py
index 4d44f022..88d8c254 100644
--- a/machina/apps/forum_conversation/views.py
+++ b/machina/apps/forum_conversation/views.py
@@ -492,8 +492,11 @@ class BaseTopicFormView(BasePostFormView):

     def form_valid(self, post_form, attachment_formset, poll_option_formset, **kwargs):
         """ Processes valid forms. """
-        save_poll_option_formset = poll_option_formset is not None \
+        save_poll_option_formset = (
+            poll_option_formset is not None
             and not self.preview
+            and kwargs['poll_options_validated']
+        )

         valid = super().form_valid(
             post_form, attachment_formset, poll_option_formset=poll_option_formset, **kwargs)
```

Reply:
Hello Ellmetha, sorry for the long time. 
Yes you are right, it work fine like that. Unfortunately any error message is returned, and i fail to achieve that.. 
I will try again but however the must important thing was to fix this little bug, so good job and thank you